common/src/main/java/org/broadleafcommerce/common/email/dao/EmailReportingDaoImpl.java

```diff
@@ -61,6 +69,25 @@ public EmailTarget createTarget() {
         return target;
     }
 
+    @Override
+    public void clearAllRecordsOlderThan(LocalDateTime date) {
+        CriteriaBuilder builder = em.getCriteriaBuilder();
+        CriteriaQuery<EmailTracking> criteria = builder.createQuery(EmailTracking.class);
+        Root<EmailTrackingImpl> EmailTrackingRoot = criteria.from(EmailTrackingImpl.class);
+
+        criteria.select(EmailTrackingRoot);
+
+        Query query = em.createQuery(criteria);
+        query.setHint(QueryHints.HINT_CACHEABLE, true);
+
+        for (Object emailTracking : query.getResultList()) {
```

Comment:
I think potentially we can have large number of records here, so this is not optimal
I think we can do this via JPQL ?
https://vladmihalcea.com/bulk-update-delete-jpa-hibernate/

Reply:
updated